cmd/juju/commands/switch.go

```diff
@@ -232,3 +319,44 @@ func (c *switchCommand) name(store jujuclient.ModelGetter, controllerName string
 	}
 	return fmt.Sprintf("%s (controller)", controllerName), nil
 }
+
+func (c *switchCommand) trySwitchToController(store jujuclient.ClientStore, controller string) (string, error) {
+	// Check that the controller actually exists
+	_, err := store.ControllerByName(controller)
+	if err != nil {
+		// If something get wrong
+		return "", errors.Trace(err)
+	}
+	targetName, err := c.name(store, controller, false)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	return targetName, errors.Trace(store.SetCurrentController(controller))
+}
+
+func (c *switchCommand) trySwitchToModel(store modelcmd.QualifyingClientStore, controller string, model string) (string, error) {
+	if err := store.SetCurrentController(controller); err != nil {
+		return "", errors.Trace(err)
+	}
```

Comment:
Why don't we check the controller exists like in `trySwitchToController`?

Reply:
It is not necessary since 
`modelName, err := store.QualifiedModelName(controller, model)` will fails with a pretty accurate message if the controller doesn't exist:

```sh
❯ juju switch jack:shaw
ERROR controller jack not found
```